.buildkite/pull-requests.json

```diff
@@ -8,8 +8,8 @@
       "allowed_list": ["renovate[bot]"],
       "build_on_commit": true,
       "build_on_comment": true,
-      "trigger_comment_regex": "^(?:(?:buildkite\\W+)?(?:build|test)\\W+(?:this|it))",
```

Comment:
I will revert it. I started to use the `build.env("GITHUB_PR_TRIGGER_COMMENT")` in conditionals (`if`) to be able to test a release and then finally I use the commit message (`build.message`).  I'm still unsure of what we want to be able to do but this can be adjusted later with the refactoring of the e2e tests.

Currently, we can trigger a pr build with `buildkite test this` or `buildkite build this`.

I added the ability to test the release steps that are not executed for a pr build but only for merge-main and tag builds.
```
buildkite test release
buildkite test e2e/main
buildkite test e2e/nightly
buildkite test release e2e/main
```

```
> grep build.message -r .buildkite | grep test
.buildkite/pipeline-build.yml:          || build.message =~ /^buildkite test .*release.*/
.buildkite/pipeline-build.yml:          || build.message =~ /^buildkite test .*release.*/
.buildkite/pipeline.yml:          || build.message =~ /^buildkite test .*e2e\/main.*/
.buildkite/pipeline.yml:          || build.message =~ /^buildkite test .*e2e\/main.*/
.buildkite/pipeline.yml:          || build.message =~ /^buildkite test .*e2e\/nightly.*/
.buildkite/pipeline.yml:      || build.message =~ /^buildkite test .*release.*/
.buildkite/pipeline-release.yml:          || build.message =~ /^buildkite test .*release.*/
.buildkite/pipeline-release.yml:          build.message =~ /^buildkite test .*release.*/
.buildkite/pipeline-release-helm.yml:          || build.message =~ /^buildkite test .*release.*/
.buildkite/pipeline-release-helm.yml:          || build.message =~ /^buildkite test .*release.*/
.buildkite/pipeline-e2e.yml:        && ( build.message !~ /^buildkite test .*e2e.*/ || build.message !~ /^buildkite test .*release.*/ )
.buildkite/pipeline-e2e.yml:      || build.message =~ /^buildkite test .*e2e\/main.*/
.buildkite/pipeline-e2e.yml:        || build.message =~ /^buildkite test .*release.*/
.buildkite/pipeline-e2e.yml:      || build.message =~ /^buildkite test .*e2e\/all.*/
```

There are other messages to run a specific part of the release (operator helm or resources helm or operatorhub):
```
> grep build.message -r .buildkite  | grep -v test
.buildkite/pipeline.yml:      || build.message =~ /^release eck/
.buildkite/pipeline-release.yml:          || build.message == "release eck on operatorhub"
.buildkite/pipeline-release-helm.yml:          || build.message == "release eck-operator helm charts"
.buildkite/pipeline-release-helm.yml:          || build.message == "release eck-resources helm charts"
.buildkite/pipeline-release-helm.yml:        if: build.message == "release eck-operator helm charts"
.buildkite/pipeline-release-helm.yml:        if: build.message == "release eck-resources helm charts"
``` 

Triggering a build via the API looks like this:
```sh
curl "https://api.buildkite.com/v2/organizations/elastic/pipelines/$pipeline/builds" -XPOST \
-H "Authorization: Bearer $BK_TOKEN" \
-d '{
    "commit": "'$commit'",
    "branch": "'$branch'",
    "message": "'"$message"'",
    "env": {
        "GITHUB_PR_TRIGGER_COMMENT": "'"$pr_comment"'"
    }
}'  
```